packages/core/src/components/size-attribute.ts

```diff
@@ -0,0 +1,82 @@
+import { DOM, customAttribute, bindingMode, inject, bindable } from 'aurelia-framework';
+import { PLATFORM } from 'aurelia-pal';
+
+export interface Size {
+  width: number;
+  height: number;
+}
+
+@customAttribute('size')
+@inject(Element)
+export class SizeCustomAttribute {
+
+  private observer: any;
+  @bindable({defaultBindingMode: bindingMode.fromView, primaryProperty: true})
+  public value: Size = {width: 0, height: 0};
+
+  @bindable({defaultBindingMode: bindingMode.fromView}) public width: number = 0;
+  @bindable({defaultBindingMode: bindingMode.fromView}) public height: number = 0;
+
+  constructor(private element: HTMLElement) {
+  }
+
+  public bind() {
+    this.observer = this.getObserver();
+    this.observer.observe(this.element);
+  }
+
+  public unbind() {
+    this.observer.disconnect();
+    this.observer = void 0;
+  }
+
+  public getObserver() {
+    if (typeof PLATFORM.global.ResizeObserver === 'function') {
+      return new PLATFORM.global.ResizeObserver((records: Array<{contentRect: DOMRectReadOnly}>) => {
+        const rect = records[0].contentRect;
+        this.sizeChanged({ width: rect.width, height: rect.height });
+      });
+    } else {
+      return new ElementSizeDirtyChecker((size) => {
+        this.sizeChanged(size);
+      });
+    }
+  }
+
+  private sizeChanged(size: Size) {
+    this.value = size;
```

Comment:
`sizeChanged` should be left to Aurelia to handle only, so `this.value` will always be equal to `size` parameter, no need to reassign here

Reply:
Good catch. I wrote this `sizeChanged` method to have a single handler between the two observers. But naming was bad because it looks like a bindable/observer handler. This was not the case as this class doesn't have any size property.

I refactored to use a `valueChanged` handler on the `value` property.